smart-contracts/contracts/cl-vault/src/helpers.rs

```diff
@@ -160,18 +160,51 @@ pub fn get_twap_price(
 //     ))
 // }
 
+const SCALE_FACTOR: u128 = 10u128.pow(12);
+
+fn scale_if_needed(
+    cur_price_sqrt: Decimal256,
+    upper_price_sqrt: Decimal256,
+    lower_price_sqrt: Decimal256,
+    current_price: Decimal256,
+) -> (bool, Decimal256, Decimal256, Decimal256, Decimal256) {
+    let scale_up_threshold = Decimal256::from_ratio(1u128, SCALE_FACTOR);
+    let product = cur_price_sqrt * upper_price_sqrt * lower_price_sqrt;
+
+    if product <= scale_up_threshold {
+        let scale_factor = Decimal256::from_atomics(SCALE_FACTOR, 0).unwrap();
+
+        // Scale the square root prices and current price
+        let scaled_cur_price_sqrt = cur_price_sqrt.checked_mul(scale_factor).unwrap();
+        let scaled_upper_price_sqrt = upper_price_sqrt.checked_mul(scale_factor).unwrap();
+        let scaled_lower_price_sqrt = lower_price_sqrt.checked_mul(scale_factor).unwrap();
+        let scaled_current_price = current_price.checked_mul(scale_factor).unwrap();
+
+        return (
```

Comment:
nit: no return statement would be needed here if the flow would be an if else, so the syntax would show not an early return but a fork in the control flow instead

Reply:
Ok I corrected the code, but instead of an if else i declared args as mutable vars and so im overriding the value only if necessary. Hope its fine